Removes lib-auth and setup lib-classifier with PJC auth

srallen opened the pull request:
Package: lib-auth, lib-classifier, app-project

Describe your changes:
This removes lib-auth, updates the bootstrap bash script, updates lib-classifier to use PJC's auth client (or oauth client), and updates app-project to remove the workaround for the mixed use of the old and new clients. 

To test it out, the dev version of the classifier: `npm run dev`
Log in and out. You'll see in the mobx dev tools a UPP is set when you sign in and reset when you sign out. 

I included one minor style fix with the footer and theme style updates that got merged today.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

rogerhutchings commented on packages/lib-classifier/src/store/UserProjectPreferencesStore.js:
```diff
@@ -64,10 +48,29 @@ const UserProjectPreferencesStore = types
+        } else {
+          self.reset()
+        }
+      } catch (e) {
```
Should be `error`

srallen commented:
I'd like to add a test for the helper function for getting the bearer token and there are broken tests related to the metadata modal (my bad for not checking in the recent PR), but I'd like to do that in a separate PR.

eatyourgreens commented:
I'm seeing MST failing to resolve references when loading the classifier on http://localhost.zooniverse.org:3000/projects/eatyourgreens/-project-testing-ground/classify I assume that's #444?

http://localhost.zooniverse.org:3000/projects/eatyourgreens/-project-testing-ground/classify?env=staging crashes the React app with an invariant violation.

eatyourgreens commented on packages/lib-classifier/src/store/UserProjectPreferencesStore.js:
```diff
@@ -64,10 +48,29 @@ const UserProjectPreferencesStore = types
+      try {
+        const bearerToken = yield getBearerToken(authClient)
+        const user = yield authClient.checkCurrent()
+        if (bearerToken && user) {
```
If there's a user, there must be a valid bearer token, so checking both is redundant here?

srallen replied:
We need the user id for the query param and a defined bearer token for making the request authenticated. It does seem redundant, but I'm not totally trusting of the PJC auth client. I don't want to make the request if for some reason either of those are undefined.

eatyourgreens commented on packages/lib-classifier/src/store/utils/getBearerToken.js:
```diff
@@ -1,10 +1,10 @@
-    if (authToken && authToken.token) {
-      return `Bearer ${authToken.token}`
+    const authToken = await authClient.checkBearerToken() || ''
+    if (authToken && authToken.access_token) {
```
I think `auth.checkBearerToken()` yields a string, not an object.

srallen replied:
That's what I thought too, but it is returning an object. At least the oauth client does. I'll check the behavior of the first party auth client. It was a mistake for me to change the default be string if it's returning an object.

eatyourgreens replied:
That's confusing. `auth.checkBearerToken()` yields a string for me in #437 (and is used that way earlier in this PR too.)
https://github.com/zooniverse/front-end-monorepo/blob/7ef5dd36d4be46b44c102a4aa59a55f3efc3faf9/packages/app-project/stores/Collections.js#L62-L63

srallen replied:
Yep, I'm thinking it's a difference between auth and oauth cilents. Even though they have the same API method, it's returning different things.

srallen replied:
I've confirmed that the auth client (first party, being used by app-project) returns a string of just the bearer token. The same method for the oauth client returns an object which includes the bearer token and expiration and the token type. So, given that the classifier is meant to be consumable by either, I'm going to have to write the helper to handle either for now.

eatyourgreens replied:
I remember now. First party handles token refresh for you, and always returns a valid token. Third party expects you to manage expired tokens, so returns an object including the expiration info.

srallen commented:
@eatyourgreens I've refactored the helper function in the classifier to be able to deal with both the first party auth and oauth clients from PJC. I added specs to make sure it's doing what we think it is.

Could you manually test with the mini dev app in the lib-classifier instead of app-project for now (also just `npm run dev` in its folder)? With the app-project, the user store isn't written to manage or track the state of the request for the user, so what's happening if you test with a project that isn't crashing is that the classifier stores initialize, including the one for the UPP, before the request for the user finishes. We should refactor this so the store is handling the async requests and tracking the state of that request. The classifier should not load until we know there is a user or not. However, that refactor is out of scope for this PR.

As for your test project, I think it is crashing because the default workflow has audio/image subjects and we haven't built the new classifier to deal with those or have a reasonable error UI yet.

eatyourgreens commented:
@srallen sounds good. I've been testing #437 with my test project, though, and not seeing errors there.
